### PR TITLE
improve TimeBucket class getTimeBucket method.

### PR DESCRIPTION
**Pull request by aderm**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [x] Improve performance

- Related issues

___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

`
    
    @Benchmark
    @BenchmarkMode(Mode.AverageTime)
    @OutputTimeUnit(TimeUnit.MICROSECONDS)
    public long testOldBucket() throws UnexpectedException {
        Calendar calendar = Calendar.getInstance();
        calendar.setTimeInMillis(time);

        long year = calendar.get(Calendar.YEAR);
        long month = calendar.get(Calendar.MONTH) + 1;
        long day = calendar.get(Calendar.DAY_OF_MONTH);
        long hour = calendar.get(Calendar.HOUR_OF_DAY);
        long minute = calendar.get(Calendar.MINUTE);
        long second = calendar.get(Calendar.SECOND);

        switch (downsampling) {
            case Second:
                return year * 10000000000L + month * 100000000 + day * 1000000 + hour * 10000 + minute * 100 + second;
            case Minute:
                return year * 100000000 + month * 1000000 + day * 10000 + hour * 100 + minute;
            case Hour:
                return year * 1000000 + month * 10000 + day * 100 + hour;
            case Day:
                return year * 10000 + month * 100 + day;
            case Month:
                return year * 100 + month;
            default:
                throw new UnexpectedException("Unknown downsampling value.");
        }
    }

    @Benchmark
    @BenchmarkMode(Mode.AverageTime)
    @OutputTimeUnit(TimeUnit.MICROSECONDS)
    public long testNewBucket() throws UnexpectedException {
        switch (downsampling) {
            case Second:
                return time;
            case Minute:
                return time / 60;
            case Hour:
                return time / (60*60);
            case Day:
                return time / (60*60*24);
            case Month:
                return time / (60*60*24*30);
            default:
                throw new UnexpectedException("Unknown downsampling value.");
        }
    }
`

WX20200109-111940@2x

### CODE REVIEW

**Comment by dmsolr** [comment]
The main point, TimeBucket is human readability.

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/TimeBucket.java [hunk 1]
```diff
@@ -17,14 +17,18 @@
...
 
 /**
  * @author peng-yongsheng
  */
 public class TimeBucket {
 
+    public static final int SECOND_BUCKET = 60;
+    public static final int HOUR_BUCKET = 60 * 60;
+    public static final int DAY_BUCKET = 60 * 60 * 24;
+    public static final int MONTH_BUCKET = 60 * 60 * 24 * 30;
```
Comment (arugal) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/TimeBucket.java]: Not every month is 30 days.

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/TimeBucket.java]: Agree, @arugal got the point, this calculation is not accurate.

Comment (aderm) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/TimeBucket.java]: It ’s really a bit problematic for the month, but it ’s not currently used, but the day, hour, and minute should be fine.

Comment (wu-sheng) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/TimeBucket.java]: Month is used for all metrics in default. Also, if you change this, all query(s) are affected. Check the timebucket usage in the query stage.

Comment (aderm) [reply oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/TimeBucket.java]: Ok, let me take a look

**Pull request closed without merging**